### PR TITLE
fix(KFLUXVNGD-876): Envtest suites bypass SetupWithManager

### PR DESCRIPTION
**Pull request by avi-biton**
Update the envtest suites to start a real controller manager
(via ctrl.NewManager + SetupWithManager + mgr.Start) so that edge-case
tests exercise the full event pipeline.

Remove all calls to reconcile and rely on the controller manager to do the work.

- Refactor `testutil` to add the infrastructure to start a real controller manager.
- Wire the manager in the BeforeSuite in `suite_test.go`.
- Update KonfluxApplicationAPIController test.
- Update KonfluxBuildServiceController test.
- Update KonfluxCertManagerController test.
- Update KonfluxDefaultTenantController test.
- Update KonfluxEnterpriseContractController test.
- Update KonfluxImageController test.
- Update KonfluxInfoController test.
- Update KonfluxIntegrationServiceController test.
- Update KonfluxInternalRegistryController test.
- Update KonfluxNamespaceListerController test.
- Update KonfluxRBACController test.
- Update KonfluxReleaseServiceController test.
- Update KonfluxSegmentBridgeController test.
- Update KonfluxUIController test.
- Update KonfluxController test.
- Update KonfluxCLIController test.

Assisted-By: Cursor

### CODE REVIEW

File: operator/internal/controller/testutil/testutil.go [hunk 1]
```diff
@@ -178,3 +188,65 @@ func FindContainer(containers []corev1.Container, name string) *corev1.Container
...
+// lifecycle should be shorter than the suite (e.g. stop them via DeferCleanup when
+// different tests need different reconciler configurations).
+func StartManagerWithContext(ctx context.Context, mgr ctrl.Manager) {
+	go func() {
+		defer GinkgoRecover()
+		Expect(mgr.Start(ctx)).To(Succeed())
+	}()
+	Eventually(func() bool {
+		return mgr.GetCache().WaitForCacheSync(ctx)
+	}).WithTimeout(30 * time.Second).Should(BeTrue())
```
[the remark this answers is not in the document]
Comment (avi-biton) [reply operator/internal/controller/testutil/testutil.go]: This issue was addressed and fixed.

**Comment by yftacherzog** [comment]
It's a lot to review manually, but trying to understand the general points of weakness, I'd say:

1. Seems as we potentially lose some more direct assertions with the new approach. I think that in time, if we get regressions that we realize could have been caught by such assertions, we could add back specific test cases that assert directly rather than using the manager, but generally, for now we should be able to ok (maybe can ask one of the agent to spot any obvious assertions that we lost and worth adding back).
2. All comments regarding concurrency, shared resources and race conditions -- we need to address those.

**Comment by avi-biton** [comment]
> It's a lot to review manually, but trying to understand the general points of weakness, I'd say:
> 
> 1. Seems as we potentially lose some more direct assertions with the new approach. I think that in time, if we get regressions that we realize could have been caught by such assertions, we could add back specific test cases that assert directly rather than using the manager, but generally, for now we should be able to ok (maybe can ask one of the agent to spot any obvious assertions that we lost and worth adding back).
> 2. All comments regarding concurrency, shared resources and race conditions -- we need to address those.

1. I added more assertions, which should cover what we had
2. All comments were addressed

**Review by yftacherzog — APPROVED** [review]
